src/nectarchain/calibration/container/waveforms.py

```diff
@@ -136,138 +157,196 @@ def check_events(self):
-            self.event_type[i]=event.trigger.event_type.value
-            self.ucts_busy_counter[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_busy_counter
-            self.ucts_event_counter[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.ucts_event_counter
+        wfs_hg_tmp = np.zeros((self.npixels, self.nsamples), dtype=np.uint16)
+        wfs_lg_tmp = np.zeros((self.npixels, self.nsamples), dtype=np.uint16)
 
+        # n_traited_events = 0
+        event_start, event_end = start, end
+        print(event_start, event_end)
 
-            self.trig_pattern_all[i] = event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.trigger_pattern.T
-
-            for pix in range(self.npixels):
-                wfs_lg_tmp[pix]=event.r0.tel[0].waveform[1,self.pixels_id[pix]]
-                wfs_hg_tmp[pix]=event.r0.tel[0].waveform[0,self.pixels_id[pix]]
+        for i, event in enumerate(self.__reader):
+            if i % 5000 == 0:
+                log.info(f"reading event number {i}")
 
-            self.wfs_hg[i] = wfs_hg_tmp
-            self.wfs_lg[i] = wfs_lg_tmp
+            if (
+                event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.event_id
+                >= event_start
+                and event.nectarcam.tel[WaveformsContainer.TEL_ID].evt.event_id
+                <= event_end
+            ):
+                self.event_id[i] = np.uint16(event.index.event_id)
+                self.ucts_timestamp[i] = event.nectarcam.tel[
+                    WaveformsContainer.TEL_ID
+                ].evt.ucts_timestamp
+                self.event_type[i] = event.trigger.event_type.value
+                self.ucts_busy_counter[i] = event.nectarcam.tel[
+                    WaveformsContainer.TEL_ID
+                ].evt.ucts_busy_counter
+                self.ucts_event_counter[i] = event.nectarcam.tel[
+                    WaveformsContainer.TEL_ID
+                ].evt.ucts_event_counter
+
+                self.trig_pattern_all[i] = event.nectarcam.tel[
+                    WaveformsContainer.TEL_ID
+                ].evt.trigger_pattern.T
+
+                for pix in range(self.npixels):
+                    wfs_lg_tmp[pix] = event.r0.tel[0].waveform[1, self.pixels_id[pix]]
+                    wfs_hg_tmp[pix] = event.r0.tel[0].waveform[0, self.pixels_id[pix]]
+
+                self.wfs_hg[i] = wfs_hg_tmp
+                self.wfs_lg[i] = wfs_lg_tmp
 
         self.__compute_broken_pixels()
 
-        #if compute_trigger_patern and np.max(self.trig_pattern) == 0:
+        # if compute_trigger_patern and np.max(self.trig_pattern) == 0:
         #    self.compute_trigger_patern()
 
-
-    def write(self,path : str, **kwargs) : 
-        """method to write in an output FITS file the WaveformsContainer. Two files are created, one FITS representing the data
-        and one HDF5 file representing the subarray configuration
+    def write(self, start, end, bloc, path: str, **kwargs):
```

Comment:
Here also, `start`, `end`, and `bloc` (`block` ?) should default to the whole run.